### PR TITLE
Cleanup test helpers

### PR DESCRIPTION
**Pull request by DDvO**
This is a follow-up on #13552, doing some test cleanup:
* move helper `.c` and `.h` files to new `test/helpers/` directory
* remove obsolete files `test/drbg_cavs_data.h` and `test/drbg_extra_test.h`
* `endecode_test.c`: Add warning that 512-bit DH key size is for testing only, as requested in https://github.com/openssl/openssl/pull/13552#pullrequestreview-540469857

### CODE REVIEW

File: test/helpers/pkcs12.c [hunk 1]
```diff
@@ -18,8 +18,8 @@
 #include <openssl/x509v3.h>
 #include <openssl/pem.h>
 
-#include "testutil.h"
-#include "pkcs12_helper.h"
+#include "../testutil.h"
+#include "./pkcs12.h"
```
Comment (levitte) [inline test/helpers/pkcs12.c:22]: Was `./` necessary?

**Review by levitte — APPROVED** [review]
Nice cleanup!

File: test/helpers/pkcs12.c (hunk 1 above)
Comment (DDvO) [reply test/helpers/pkcs12.c:22]: It turned out not to be necessary, but I think it's good to make clear that this `#include` is not for `include/openssl/pkcs12.h`-

**Comment by DDvO** [comment]
> Nice cleanup!

Thanks!

I'm also considering moving (in a separate PR) all `test/*.pem` to `test/certs`
and maybe renaming `test/certs` to `test/creds` because it contains not only certs but also many keys and some CSRs and CRLs.
I could ask our working student to do that.
What do you think?

File: test/helpers/pkcs12.c (hunk 1 above)
Comment (levitte) [reply test/helpers/pkcs12.c:22]: You gotta admit that it kinda sticks out, when the rest of the source has "filename.h" without any directory part to mean the header file in the same directory as the source, `"e_os.h"` being the unique exception...

Comment (DDvO) [reply test/helpers/pkcs12.c:22]: Yes, this - purposely - points out which `pkcs12.h` is meant.
Shall I still remove the `./`?

Comment (levitte) [reply test/helpers/pkcs12.c:22]: Frankly, [shrug].  I just gave you my thoughts, and leave it to you do decide what you want to do.  For the compiler, this makes zero difference.  If you think this makes a difference for humans reading the source, then fine

Comment (DDvO) [reply test/helpers/pkcs12.c:22]: Fine, then I'd say let's keep it for clarity.

**Comment by DDvO** [comment]
And how about my suggestion of moving the `*.pem` cert/key files from `test/` into a subdir?

**Comment by richsalz** [comment]
I do not think putting all keys and certs into one directory is a good idea.  Maybe if there's duplicate files, sure, but keep per-test data with the test is better.

File: test/helpers/pkcs12.c (hunk 1 above)
Comment (richsalz) [reply test/helpers/pkcs12.c:22]: I think a comment is more clear and also less confusing.

Comment (DDvO) [reply test/helpers/pkcs12.c]: > I think a comment is more clear and also less confusing.

All right, done.

**Comment by DDvO** [comment]
> I do not think putting all keys and certs into one directory is a good idea. Maybe if there's duplicate files, sure, but keep per-test data with the test is better.

I generally agree, but the point here is different:
There is already quite a bunch of cert and key files directly in `test/`, used as test data _across_ various tests.
There are even many more such test credential files in `test/certs/`.
BTW, could well be that part of them are no longer used (and hopefully replaced by specific data in `test/recipes/*_data/*`).

So the question is where to put them such that they at least do not distract from the test source files in `test/*.c`.

**Comment by richsalz** [comment]
> So the question is where to put them such that they at least do not distract from the test source files in test/*.c.

As a downstream consumer of OpenSSL, I'd like to encourage you to work on things that the project already listed as required for 3.0

**Comment by DDvO** [comment]
Good point, that type of cleanup really is not urgent.

**Comment by DDvO** [comment]
Merged - thanks

**Pull request closed without merging**